stream-cipher/src/lib.rs

```diff
@@ -94,13 +95,38 @@ pub trait SyncStreamCipher {
     fn try_apply_keystream(&mut self, data: &mut [u8]) -> Result<(), LoopError>;
 }
 
-/// Synchronous stream cipher seeking trait.
+/// Trait for seekable stream ciphers.
+///
+/// Methods of this trait are generic over the [`NumSeek`] trait, which is
+/// implemented for the following numeric types: `u8`, `u16`, `u32`, `i32`,
+/// `u64`, and `u128`.
 pub trait SyncStreamCipherSeek {
-    /// Return current position of a keystream in bytes from the beginning.
-    fn current_pos(&self) -> u64;
+    /// Try to get current keystream position
+    ///
+    /// Returns [`LoopError`] if position can not be represented by type `T`
+    fn try_get_pos<T: SeekNum>(&self) -> Result<T, OverflowError>;
+
+    /// Try to seek to the given position
+    ///
+    /// Returns [`LoopError`] if provided position value is bigger than
+    /// keystream leangth
+    fn try_seek<T: SeekNum>(&mut self, pos: T) -> Result<(), LoopError>;
+
+    /// Get current keystream position
+    ///
+    /// # Panics
+    /// If position can not be represented by type `T`
+    fn get_pos<T: SeekNum>(&self) -> T {
```

Comment:
Bikeshed, but: I'd prefer `current_pos` or `pos` here. [It's generally considered not idiomatic to name getter methods `get_*` in Rust](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter)